hellabot.go

```diff
@@ -118,8 +118,12 @@ func (bot *Bot) handleIncomingMessages() {
 		bot.con.SetDeadline(time.Now().Add(bot.PingTimeout))
 		msg := ParseMessage(scan.Text())
 		bot.Debug("Incoming", "raw", scan.Text(), "msg.To", msg.To, "msg.From", msg.From, "msg.Params", msg.Params, "msg.Trailing", msg.Trailing)
+		// Otherwise this is mutable by DropTrigger and AddTrigger
 		bot.triggersMu.Lock()
-		triggers := bot.triggers
+		triggers := make([]Trigger, 0, len(b.triggers))
```

Comment:
You can use the builtin `copy` function to do this.

Reply:
Will `copy()` create a new array?